tests/functional/codegen/features/iteration/test_for_range.py

```diff
@@ -228,39 +228,13 @@ def foo(a: {typ}) -> {typ}:
     assert c.foo(100) == 31337
 
 
-# test that we can get to the upper range of an integer
-@pytest.mark.parametrize("typ", ["uint8", "int128", "uint256"])
-def test_for_range_edge(get_contract, typ):
-    code = f"""
-@external
-def test():
-    found: bool = False
-    x: {typ} = max_value({typ})
-    for i in range(x, x + 1):
-        if i == max_value({typ}):
-            found = True
-
-    assert found
-
-    found = False
-    x = max_value({typ}) - 1
-    for i in range(x, x + 2):
-        if i == max_value({typ}):
-            found = True
-
-    assert found
-    """
-    c = get_contract(code)
-    c.test()
```

Comment:
are we sure we should remove this test? is it superseded by some other test?

Reply:
Without the special handling for the `range(x, x+N)`, `max + 1` will overflow.
Having a range that goes until the edge is not generally useful, since arrays are 0-indexed.
In order to not overflow, the `i` must be calculated later on
```
    for i in range(x - 1, x):
        if i + 1 == max_value({typ}):
            found = True
```